components/tikv_util/src/lib.rs

```diff
@@ -447,7 +447,9 @@ impl<T> DerefMut for MustConsumeVec<T> {
 
 impl<T> Drop for MustConsumeVec<T> {
     fn drop(&mut self) {
-        if !self.is_empty() {
+        // Double-panic should be prevented, otherwise it will abort.
+        // Resource leak on panic is normal, so we do not even log here.
+        if !self.is_empty() && !thread::panicking() {
```

Comment:
em, can we provide another macro like safe_panic to avoid double panic? 

if we want to use in other places, we still have to first check panicking, then do the panic.

Reply:
I commented under #4202 a few days earlier. Do you think it's possible that with minidump we can set panic=abort and just no longer deal with such a problem?

Reply:
maybe, but I still think we can provide it because we don' use minidump in our developments.

Reply:
I agree that the dtors should be written correctly regardless of the unwinding strategy.